oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -111,66 +115,58 @@
-   * @param projectId the project used for billing
-   * @param quotaProjectId the project used for quota and billing purposes. May be null.
-   * @param lifetime number of seconds the access token should be valid for. The value should be at
-   *     most 43200 (12 hours). If the token is used for calling a Google API, then the value should
-   *     be at most 3600 (1 hour). If the given value is 0, then the default value 3600 will be used
-   *     when creating the credentials.
-   * @param useJwtAccessWithScope whether self signed JWT with scopes should be always used.
+   * @param builder A builder for {@link ServiceAccountCredentials} See {@link
+   *     ServiceAccountCredentials.Builder} List of fields being set: clientId: client ID of the
+   *     service account from the console. May be null. clientEmail: client email address of the
+   *     service account from the console privateKey: RSA private key object for the service account
+   *     privateKeyId: private key identifier for the service account. May be null. scopes: scope
+   *     strings for the APIs to be called. May be null or an empty collection. defaultScopes:
+   *     default scope strings for the APIs to be called. May be null or an empty. transportFactory:
+   *     HTTP transport factory, creates the transport used to get access tokens. tokenServerUri:
+   *     URI of the end point that provides tokens. serviceAccountUser: email of the user account to
+   *     impersonate, if delegating domain-wide authority to the service account. projectId: the
+   *     project used for billing quotaProjectId: the project used for quota and billing purposes.
+   *     May be null. lifetime: number of seconds the access token should be valid for. The value
+   *     should be at most 43200 (12 hours). If the token is used for calling a Google API, then the
+   *     value should be at most 3600 (1 hour). If the given value is 0, then the default value 3600
+   *     will be used when creating the credentials. useJwtAccessWithScope: whether self signed JWT
+   *     with scopes should be always used. defaultRetriesEnabled: whether default retries are
+   *     enabled
    */
-  ServiceAccountCredentials(
-      String clientId,
-      String clientEmail,
-      PrivateKey privateKey,
-      String privateKeyId,
-      Collection<String> scopes,
-      Collection<String> defaultScopes,
-      HttpTransportFactory transportFactory,
-      URI tokenServerUri,
-      String serviceAccountUser,
-      String projectId,
-      String quotaProjectId,
-      int lifetime,
-      boolean useJwtAccessWithScope) {
-    this.clientId = clientId;
-    this.clientEmail = Preconditions.checkNotNull(clientEmail);
-    this.privateKey = Preconditions.checkNotNull(privateKey);
-    this.privateKeyId = privateKeyId;
-    this.scopes = (scopes == null) ? ImmutableSet.<String>of() : ImmutableSet.copyOf(scopes);
+  ServiceAccountCredentials(ServiceAccountCredentials.Builder builder) {
```

Comment:
I responded yes when you asked about this yesterday from my phone.  And I'm ok w/ it if you insist, but really the `.build()` method should return a `ServiceAccountCredential`.  Your choice. see go/ej3e-2 (final few paragraphs)